core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -155,6 +155,23 @@ object Partition extends KafkaMetricsGroup {
   }
 }
 
+
+sealed trait AssignmentState {
+  def replicas: Seq[Int]
+  def replicationFactor: Int = replicas.size
+  def isAddingReplica(brokerId: Int): Boolean = false
+}
+
+case class OngoingReassignmentState(addingReplicas: Seq[Int],
+                                    removingReplicas: Seq[Int],
+                                    replicas: Seq[Int]) extends AssignmentState {
+
+  override def replicationFactor: Int = replicas.diff(addingReplicas).size // keep the size of the original replicas
```

Comment:
neat `diff` method! I didn't know about this when implementing `PartitionReplicaAssignment`

Reply:
Well, scala is always full of surprises :)